icicle/tests/test_hash_api.cpp

```diff
@@ -590,31 +591,158 @@ TEST_F(HashApiTest, MerkleTreeExample)
 }
 
 #ifdef POSEIDON
+// p = 0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000001
 
   #include "icicle/fields/field_config.h"
+  #include "poseidon/constants/bn254_poseidon.h"
+
 using namespace field_config;
+using namespace poseidon_constants_bn254;
 
   #include "icicle/hash/poseidon.h"
 
-TEST_F(HashApiTest, poseidon12)
+TEST_F(HashApiTest, poseidon12_single_hash)
 {
-  const uint64_t arity = 12; // Number of input elements
+  const unsigned t = 12;
+  auto config = default_hash_config();
+
+  auto input = std::make_unique<scalar_t[]>(t);
+  scalar_t::rand_host_many(input.get(), t);
+  config.batch = 1;
+
+  auto run = [&](const std::string& dev_type, scalar_t* out, bool measure, const char* msg, int iters) {
+    Device dev = {dev_type, 0};
+    icicle_set_device(dev);
+
+    std::ostringstream oss;
+    oss << dev_type << " " << msg;
+
+    auto poseidon = Poseidon::create<scalar_t>(t);
+
+    START_TIMER(POSEIDON_sync)
+    for (int i = 0; i < iters; ++i) {
+      ICICLE_CHECK(poseidon.hash(input.get(), t, config, out));
+    }
+    END_TIMER(POSEIDON_sync, oss.str().c_str(), measure);
+  };
+
+  auto output_cpu = std::make_unique<scalar_t[]>(config.batch);
+  auto output_cuda = std::make_unique<scalar_t[]>(config.batch);
+
+  run(s_reference_target, output_cpu.get(), VERBOSE /*=measure*/, "poseidon", ITERS);
+  run(s_main_target, output_cuda.get(), VERBOSE /*=measure*/, "poseidon", ITERS);
+
+  ASSERT_EQ(0, memcmp(output_cpu.get(), output_cuda.get(), config.batch * sizeof(scalar_t)));
+}
+
+// TEST_F(HashApiTest, poseidon3_single_hash_domain_tag)
```

Comment:
should we remove this test or uncomment it ? now domain tag works but note that it's tested below in poseidon3_batch and also in rust all arities (3,5,9,12) are tested with and without domain tag.

Reply:
I'll remove it for now

Reply:
Now this test causes a segmentation fault and should be debugged.
So for now I'll remove it  and if there will be time before 3.1 release I'll debug it